brave/browser-laptop
# Rework alert/confirm to be tab-modal (instead of application-modal)

## Test Plan
Testing alert
1. Visit http://jsbin.com/fiyojusahu/edit?html,output
2. In the `output` area, click the `Click me to test an alert` button
3. Confirm the alert works as expected

Testing confirm 
1. Visit http://jsbin.com/sadunogefu/edit?html,output
2. In the `output` area, click the `Click me to test a confirm` button
3. Confirm the confirm works as expected (try clicking OK and also cancel)

See issues for issue specific tests:
#3794 - Hard to exit/close Brave when site spams you with message box / alerts
#2755 - prevent js alert spoofing attacks
#6901 - alert() popups should appear below tabs
#7213 - Brave crashed on macOS 10.12.3 via.window.open()
#7280 - Invalid sync code alert needs proper heading

## Automated tests
```bash
# unit tests
npm run unittest -- --grep="MessageBox component unit tests"
npm run unittest -- --grep="tabMessageBoxState unit tests"
npm run unittest -- --grep="tabMessageBox unit tests"
npm run unittest -- --grep="tabState unit tests"
npm run unittest -- --grep="Main component unit tests"
npm run unittest -- --grep="NavigationBar component unit tests"
npm run unittest -- --grep="UrlBar component unit tests"
npm run unittest -- --grep="UrlBarIcon component unit tests"

# webdriver tests
npm run uitest -- --grep="MessageBox component tests"
```


Rework alert/confirm to be tab-modal (instead of application-modal)

Message box:
- shows an on/off switch for "Prevent this page from creating additional dialogs". Fixes https://github.com/brave/browser-laptop/issues/3794
- does not include Brave logo; shows origin; does not need to show "switch to tab" because it's already bound to tab. Fixes https://github.com/brave/browser-laptop/issues/2755
- is shown BELOW tabs now. Fixes https://github.com/brave/browser-laptop/issues/6901

Other changes include:
- Removed "An embedded page at " from source message (shown on every message)
- Rename showMessageBox/hideMessageBox/clearMessageBoxes actions to showNotification/hideNotification/clearNotification
- Converted most styles to Aphrodite

Auditors: @bridiver, @diracdeltas
Folks who may also be interested: @bbondy, @darkdh

## Special note
Electron [recently received a PR to implement an optional checkbox](https://github.com/electron/electron/pull/8590) in electron.dialog.showMessageBox... but this is a better choice for us because it's tab-modal (you can switch tabs while the alert is up, etc).

## Screenshots

![screen shot 2017-02-06 at 10 32 02 am](https://cloud.githubusercontent.com/assets/4733304/22658517/9ec24df2-ec57-11e6-83e0-2f08b1edbea8.png)

![screen shot 2017-02-06 at 10 30 04 am](https://cloud.githubusercontent.com/assets/4733304/22658525/a6e94c92-ec57-11e6-8cdf-8c8c6c5c75ce.png)

![screen shot 2017-02-06 at 10 27 50 am](https://cloud.githubusercontent.com/assets/4733304/22658530/ac629598-ec57-11e6-825d-db7d5ad5ab6a.png)

![screen shot 2017-02-06 at 11 48 20 am](https://cloud.githubusercontent.com/assets/4733304/22661351/4b28d98a-ec62-11e6-8a0c-bc9f603fb1a8.png)

cc: @ayumi
